balancer_wrapper.go

```diff
@@ -279,6 +286,21 @@ func (acbw *acBalancerWrapper) updateState(s connectivity.State, curAddr resolve
 		if s == connectivity.Ready {
 			setConnectedAddress(&scs, curAddr)
 		}
+		// Invalidate the health listener by changing the state.
+		acbw.connectivityState.Store(&scs.ConnectivityState)
```

Comment:
The description above is correct. I've added a doc comment for `RegisterHealthListener` explaining that the presently registered health listener is used as long as the stored connectivity state is not updated.

I started off storing the `health listener + connectivity state`. Then I realised that it was not possible to compare listener functions for equality to determine if a new listener was registered. I ended up storing  only a pointer to the connectivity state and getting rid of the health listener since it became redundant. The reason for supporting only one listener is to simplify the client side health checking producer and allow un-registering the listener by registering a `nil` listener.